### ISSUE TITLE
[Bug] Some functions of the hardware mouse cannot be used.

### ISSUE THREAD

**Issue by yichen-hub** [PojavLauncherTeam/PojavLauncher_iOS#616]
### Describe the bug

After connecting the hardware mouse, you can't use the mouse to move the viewing angle, and you can't use all the functions of right mouse button.

### Steps To Reproduce

```markdown
Start pojavlauncher and connect the hardware mouse through Bluetooth.
```

### Expected Behavior

I hope the mouse can work properly

### Platform

```markdown
- Device model: iphone13 mini
- iOS version:16.2 
- PojavLauncher version: 2.2
```

**Comment by GaucovyAero** [comment]
what mouse are you using (exact model) and how are you connecting it?

**Comment by yichen-hub** [comment]
ugreen mu002
Connect via Bluetooth

**Comment by yichen-hub** [comment]
UGREEN MU002
Bluetooth connection

**Comment by GaucovyAero** [comment]
hmm okay, thanks for the info i guess

**Comment by yichen-hub** [comment]
I'm sorry to bother you, but I want to ask if there is any way to fix this problem?

**Comment by GaucovyAero** [comment]
> I'm sorry to bother you, but I want to ask if there is any way to fix this problem?

well im not a developer and i dont have an iOS device to test on (for now), so i cant really tell you whether its an issue with iOS compatibility with your mouse or Pojav itself, or whether we can even fix it

**Comment by yichen-hub** [comment]
Thank you very much for your reply. I can provide more information if necessary.

**Comment by Diskhdjsksk** [comment]
Dont sent to me this go away

**Comment by YOUTH0890** [comment]
Your problem is the same as mine.

**Comment by ItAnthon** [comment]
can you send over a log so that i can determine what build you’re on?

**Comment by Zhou-Shilin** [comment]
fix: use the touch screen to start the game by clicking on "Start". Keep the cursor in the center of the screen until the game enters the loading screen. This works for me.
I think the problem is that Pojavlauncher hides your iOS cursor after starting the game and displays it as the same cursor as it looks on a PC, but your ios cursor is only hidden and not really gone, which leads to the mouse not working because if the cursor is on a custom control before starting the game, clicking on the cursor=clicks on that control, moving the mouse=moving the hidden cursor(which is not the game cursor)

**Comment by ItAnthon** [comment]
@Zhou-Shilin another thing you could do is swipe up (don’t close PojavLauncher fully) and open the app using the iOS cursor. This fixes everything. This is a weird bug for sure, but it’ll likely take a bit to get fixed since it’s not breaking the launcher, and ofc has a couple of work-arounds.

**Comment by ItAnthon** [comment]
@yichen-hub it actually seems that Bluetooth mice may have an issue, but id still have to test and reproduce the issue and bring it up to the devs. 

can you tell me what exactly doesn’t work on Bluetooth mice? (Literally tell me everything)

**Comment by ItAnthon** [comment]
@yichen-hub today I tested a Bluetooth mouse. 

This mouse worked as intended, (although it was laggy cause of Bluetooth of course) even the side buttons worked, which I didn’t expect. If you haven’t already, please use the latest github actions build. 

If curious: iPadOS 16.6.1, TrollStore, Latest PojavLauncher GitHub actions build.

**Comment by Tsuser1** [comment]
Replicated issues using hardware mice on the latest development iOS version of PojavLauncher. The PojavLauncher seems to ignore the presence of a mouse, and moving the mouse around has no effect. Attempted connecting the mouse prior to starting the game, during the game, and after opening the app switch plane, this had no effect.

```
[Pre-Init] Version: 3.0-development
[Pre-Init] Commit: 3f0ec78 (main)
[Pre-Init] Device: iPhone 15 Pro
[Pre-Init] iOS 17.4.1 (21E236) (Unjailbroken)
```

Attempted using several mice, ranging from a wired dumb mouse, a wireless mouse with a USB dongle, and a Bluetooth mouse. All three mice exhibited the same behavior. Enabling AssistiveTouch allowed me to use iOS to move around the virtual mouse, but this isn’t a good solution. It does however make me believe this is a PojabLauncher bug, and not an iOS bug since it worked for that.

**Comment by Kiokingz** [comment]
> Replicated issues using hardware mice on the latest development iOS version of PojavLauncher. The PojavLauncher seems to ignore the presence of a mouse, and moving the mouse around has no effect. Attempted connecting the mouse prior to starting the game, during the game, and after opening the app switch plane, this had no effect.
> 
> ```
> [Pre-Init] Version: 3.0-development
> [Pre-Init] Commit: 3f0ec78 (main)
> [Pre-Init] Device: iPhone 15 Pro
> [Pre-Init] iOS 17.4.1 (21E236) (Unjailbroken)
> ```
> 
> Attempted using several mice, ranging from a wired dumb mouse, a wireless mouse with a USB dongle, and a Bluetooth mouse. All three mice exhibited the same behavior. Enabling AssistiveTouch allowed me to use iOS to move around the virtual mouse, but this isn’t a good solution. It does however make me believe this is a PojavLauncher bug, and not an iOS bug since it worked for that.

do you have "Unlock iPadOS UI" setting on? that setting is what lets you use mouse, keyboard, etc.
if you haven't had it on, please enable it in pojavlauncher settings

**Comment by Kiokingz** [comment]
looks like i didn't read your message properly, but mouse somewhat works for me with no issues

**Comment by Tsuser1** [comment]
I just checked to make sure, and yes, I do have the “Unlock iPadOS UI” setting enabled. None of my mice are working (Bluetooth or USB), so I’m not really sure what else to try. The log entries lead me to believe the mouse is seen, but something else is conflicting with it working.

**Comment by Tsuser1** [comment]
Yes, I can use the mice normally on a computer, or elsewhere in iOS by enabling AssistiveTouch.
